user-interface/src/main/java/life/qbic/datamanager/views/projects/project/measurements/MeasurementDetailsComponent.java

```diff
@@ -590,4 +613,49 @@ private Span pooledMeasurementEntry(String propertyLabel, String propertyValue)
     }
   }
 
+  public static class MeasurementTechnologyTab extends Tab {
+
+    private final Span countBadge;
+    private final Span technologyNameComponent;
+    private final String technology;
+
+    public MeasurementTechnologyTab(String technology, int measurementCount) {
+      this.technology = technology;
+      technologyNameComponent = new Span();
+      this.countBadge = createBadge();
+      Span sampleCountComponent = new Span();
+      sampleCountComponent.add(countBadge);
+      this.add(technologyNameComponent, sampleCountComponent);
+      setTechnologyName(technology);
+      setMeasurementCount(measurementCount);
+    }
+
+    public String getTabLabel() {
+      return technology;
+    }
+
+    /**
+     * Helper method for creating a badge.
+     */
+    private static Span createBadge() {
+      Tag tag = new Tag(String.valueOf(0));
+      tag.setTagColor(TagColor.CONTRAST);
```

Comment:
Shouldn't the coloring be handled by CSS? What am I missing?

Reply:
we always use TagColor like that, I'm not sure if that's necessary, though